code/modules/mob/living/say.dm

```diff
@@ -329,7 +329,12 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	return 1
 
 /mob/living/proc/can_speak_vocal(message) //Check AFTER handling of xeno and ling channels
-	if(HAS_TRAIT(src, TRAIT_MUTE) && get_message_language(message) != /datum/language/signlanguage)
+	var/obj/item/bodypart/leftarm = get_bodypart(BODY_ZONE_L_ARM)
+	var/obj/item/bodypart/rightarm = get_bodypart(BODY_ZONE_R_ARM)
+	if(HAS_TRAIT(src, TRAIT_MUTE) && get_selected_language() != /datum/language/signlanguage)
+		return 0
+			
+	if (rightarm.is_disabled() && leftarm.is_disabled() && get_selected_language() == /datum/language/signlanguage)
```

Comment:
this should actually probably check if leftarm and rightarm are not null and return early if they are, since it's possible if the mob has no arm for get_bodypart to return null.

Reply:
Fair enough. Will add that.